### DISCUSSION TITLE
Does all mutations require explicit custom update functions to update the cache?

### DISCUSSION THREAD

**Discussion by Axedyson** [urql-graphql/urql#2547] [Q&A] [answered]
I'm a bit confused of when to use custom update functions or not for mutations to update the cache. Could you explain the different general scenarios of using custom update functions and not for mutations? What I've gathered so far is that mutations require a custom update function to update the cache when the mutation is mutating a list like adding or removing and item, or if the mutation just simply creates a new object that other queries may need. It seems like - though I haven't experienced it yet - that the urql graphcache only updates the cache for mutations when the mutation is just altering/updating an entity not creating or deleting it

**Comment by JoviDeCroock** [comment, marked as answer]
As explained in the [docs](https://formidable.com/open-source/urql/docs/graphcache/normalized-caching/#non-automatic-relations-and-updates) when the result selection-set does not tell us what to update or how to update it we won't be able to.

i.e. if you return an updated entity that is in cache we will be able to just update the fields if they are requested in the result

if you create/delete an entity more often than not we won't know what to do so as a general rule of thumb linking and unlinking will frequently require you to do a custom update while an update to an entity should happen automatically as long as you request the fields.

Which makes sense as when you create an entity we don't know where to put it, does `Query.x(id: y)` return it or is it a list or how are they linking, ... similar to delete we don't know that the action was delete, else we would have to know what happens in your back-end

**Reply by Axedyson** [reply]
Insane reply speed, thank you! I'll make sure to read the docs more thoroughly next time!
